src/core/providers/ogr/qgsgeopackageproviderconnection.cpp

```diff
@@ -168,6 +174,11 @@ void QgsGeoPackageProviderConnection::renameVectorTable( const QString &schema,
   }
 }
 
+QgsVectorLayer *QgsGeoPackageProviderConnection::createSqlVectorLayer( const QgsAbstractDatabaseProviderConnection::SqlVectorLayerOptions &options ) const
+{
+  return new QgsVectorLayer( QStringLiteral( "%1|subset=%2" ).arg( uri(), options.sql ), options.layerName.isEmpty() ? QStringLiteral( "QueryLayer" ) : options.layerName, providerKey() );
```

Comment:
Would be safer here to decode uri(), add the subset value into the QVariantMap, and then re-encode the uri. encodeUri has a lot of complex logic to handle this kind of thing, so it's safer to rely on that vs the hardcoded string version.